File: backend/app/schemas/importacao.py
from datetime import date
from typing import List

from pydantic import BaseModel


class ImportacaoDuplicata(BaseModel):
    descricao: str
    valor: float
    tipo: str
    data: date


class ImportacaoErro(BaseModel):
    indice: int
    descricao: str
    motivo: str


class ImportacaoResult(BaseModel):
    formato_detectado: str
    total_no_arquivo: int
    importadas: int
    duplicatas: List[ImportacaoDuplicata]
    erros: List[ImportacaoErro]

File: frontend/src/types/importacao.ts
export interface ImportacaoDuplicata {
  descricao: string
  valor: number
  tipo: string
  data: string
}

export interface ImportacaoErro {
  indice: number
  descricao: string
  motivo: string
}

export interface ImportacaoResult {
  formato_detectado: string
  total_no_arquivo: number
  importadas: number
  duplicatas: ImportacaoDuplicata[]
  erros: ImportacaoErro[]
}
